### ISSUE TITLE
Dice and Jaccard score not improving

### ISSUE THREAD

**Issue by reachsak** [nibtehaz/MultiResUNet#8]
Hi @nibtehaz , I've been trying your model for my image segmentation projects (with same settings) but with 365 images which split (0.2) into 292 for training and 73 for testing and manage to get a fairly good results of the dice and Jacard score. But I found out that the jacard score is still at around 0.7 for too many epochs (20-30 epochs) and stop increasing even after 100 epochs. And i found out among those 73 testing images , for some predicted image the jacard score did go up to 0.91 for some only for 0.65 which totally average to around 0.7. 
I know it should be the dataset problems , I'd like if there is any way to improved the overall jacard score with that same dataset ? Should I just keep training ? Or make some changes to the dataset ?

**Comment by reachsak** [comment]
And I have one more question regarding to the calculated Jacard and dice score after each epoch. As I know there are training set , validation set and testing set. I am quite confuse in identifying the X_test and Y_test in this project to be the validation set in the training process of this model or are   the actual testing set of this model ? 
I'd like to know can we consider the jaccard and dice score calculated from the X_test and Y_test in this project  for the accuracy evaluation of the final trained model ? or we need to prepare another set of data (Test set ) to use once at the end one more time to evaluate the final trained model.
